docs/guides/migrating.md

```diff
@@ -1,15 +1,46 @@
-Academy v0.2 makes numerous breaking changes. This page highlights the most important changes to help you migrate your code from v0.1 to v0.2.
+This page documents key breaking changes and tips for migrating between beta versions of Academy.
 
 Please refer to our [Versioning Policy](../contributing/releases.md#versioning) for more details on when we make breaking changes.
 
-## Academy is now async
+## Migrating to v0.3
+
+Academy v0.3 makes a few breaking changes to improve the ergonomics of common usage patterns.
+
+### Handle actions are blocking by default
+
+Previously, invoking an action on a [`Handle`][academy.handle.Handle.action] returns a [`Future`][asyncio.Future] to the result.
+This resulted in verbose syntax when the result was immediately needed.
+```python
+future = await handle.get_count()
+result = await future
+# or
+result = await (await handle.get_count())
+```
+
+Now, action requests block and return the final result of the action.
+```python
+result = await handle.get_count()
+```
+Code that wants to submit the request and later block on it can create a [`Task`][asyncio.create_task].
```

Comment:
Would be good to add a note here on why someone might want to do this. For eg, launch several long-running actions and wait on them asynchronously with `asyncio.as_completed`.